quickannotator/api/v1/image/__init__.py

```diff
@@ -91,19 +92,67 @@ def get(self, args, project_id):
         """
         result = db.session.query(qadb.Image).filter(qadb.Image.project_id == project_id).all()
         return result, 200
+    
+@bp.route('/project/<int:project_id>', endpoint="images")
+class ImageSearch(MethodView):
+    @bp.arguments(SearchImageArgsSchema, location='query')
+    @bp.response(200, ImageRespSchema(many=True))
+    def get(self, args, project_id):
+        """     returns a list of Images
+        """
+        # images = db.session.query(qadb.Image).filter(qadb.Image.project_id == project_id).all()
+        # if images is not None:
+        #     return images
+        # else:
+        #     abort(404, message="Images not found")
+        images = db.session.query(
+            *[getattr(qadb.Image, column.name) for column in qadb.Image.__table__.columns],
+            db.func.ST_AsGeoJSON(qadb.Image.embedding_coord).label('embedding_coord')
+        ).filter(qadb.Image.project_id == project_id).all()
+        if images is not None:
+            return images, 200
+        else:
+            abort(404, message="Image not found")
 
 #################################################################################
+@bp.route('/upload/files', methods=["POST"])
+def upload_files():
+    """Upload a file"""
+    filenames=[]
+    files = request.files.getlist('files')
+    
+    for file in files:
+        filename = secure_filename(file.filename)
+        os.makedirs('./upload', exist_ok=True)
+        filepath = os.path.join('./upload',filename)
+        file.save(filepath)
+        filenames.append(filename)
+        print(filename)
+
+    return 'file uploaded successfully'
+
+
 @bp.route('/<int:image_id>/<int:file_type>/file', endpoint="file")
 class ImageFile(MethodView):
     def get(self, image_id, file_type):
         """     returns an Image file   """
         result = db.session.query(qadb.Image).filter(qadb.Image.id == image_id).first()
-
+        full_path = os.path.join(current_app.root_path, result.path)
         if file_type == 1:  # image file
-            return send_from_directory(result['path'], result['name'])
+            return send_from_directory(full_path, result.name)
         elif file_type == 2:    # thumbnail file
-            # TODO implement thumbnail file
-            pass
+            try:
+                slide = ops.OpenSlide(full_path)
+                thumbnail = slide.get_thumbnail((256, 256))
```

Comment:
we should think about the time cost here - if these images are on e.g., S3, and there are 1,000 images, is this going to take a long time? also, a 256 x 256 thumbnail is very small, so we could easily consider saving it in the project directory so that the first time its slow but all other times its very fast since it will be loaded directly from disk?